pkg/commands/kustomizationfile.go

```diff
@@ -100,3 +120,73 @@ func stringInSlice(str string, list []string) bool {
 	}
 	return false
 }
+
+func (mf *kustomizationFile) getFieldComments(content []byte) error {
+	buffer := bytes.NewBuffer(content)
+	var comments [][]byte
+	incomments := false
+	pattern := regexp.MustCompile("^(" + strings.Join(kustomizationFields, "|") + "):")
+	var err error
+	var line []byte
+	for err == nil {
+		line, err = buffer.ReadBytes('\n')
+		if strings.HasPrefix(string(line), "#") {
+			if !incomments {
+				incomments = true
+			}
+			comments = append(comments, line)
+			continue
+		}
+		if pattern.Match(line) {
+			incomments = false
+			fields := pattern.FindSubmatch(line)
+			field := fields[1]
+			mf.comments = append(mf.comments, fieldComment{field: string(field), comment: bytes.Join(comments, []byte(``))})
+			mf.fields[string(field)] = true
+			comments = [][]byte{}
+		}
+	}
+	if err != io.EOF {
+		return err
+	}
+	return nil
+}
+
+func (mf *kustomizationFile) marshal(kustomization *types.Kustomization) ([]byte, error) {
+	output := []byte{}
+	for _, comment := range mf.comments {
+		output = append(output, comment.comment...)
+		content, err := marshalField(comment.field, kustomization)
+		if err != nil {
+			return content, err
+		}
+		output = append(output, content...)
+	}
+	for field, found := range mf.fields {
+		if found {
+			continue
+		}
+		content, err := marshalField(field, kustomization)
```

Comment:
can u add a comment on what happens if the field wasn't in the original file and wasn't added?

Reply:
Sure, it returns an empty []byte. Will also add this comment in the PR.